Client/Frontend/Sync/BraveCore/Bookmarks/BookmarkManager.swift

```diff
@@ -0,0 +1,303 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Data
+import BraveCore
+import BraveShared
+import CoreData
+import Shared
+
+private let log = Logger.browserLogger
+
+class BookmarkManager: NSObject {
```

Comment:
Does this have to be a `NSObject` subclass for any reason?

Reply:
It doesnt have to be `NSObject` subclass. I generally make managers subclass of NSObject so they will conform NSObjectProtocol and in addition have them conform Equatable and Hashable but in this particular case it is not needed.

Removed at https://github.com/brave/brave-ios/pull/4024/commits/34b14995e0c684215a0690d4a1adc9bc0632cfac